interface/modules/custom_modules/oe-module-faxsms/src/Controller/ClickatellSMSClient.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * Clickatell SMS Controller
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Modules\FaxSMS\Controller;
+
+class ClickatellSMSClient extends AppDispatch
+{
+    private $appKey;
+    private $phone;
+
+    public function __construct()
+    {
+        if (empty($GLOBALS['oefax_enable_sms'] ?? null)) {
+            throw new \RuntimeException(xlt("Access denied! Module not enabled"));
+        }
+        parent::__construct();
+    }
+
+    /**
+     * @return array|mixed
+     */
+    public function getCredentials(): mixed
+    {
+        $credentials = appDispatch::getSetup();
+
+        $this->appKey = $credentials['appKey'];
+        $this->phone = $credentials['phone'];
+
+        return $credentials;
+    }
+
+    /**
+     * @return string
+     */
+    public function sendSMS(string $toPhone = '', string $subject = '', string $message = '', string $from = ''): string
+    {
+        // $subject and $from do not have valid/meaningful values passed in
+
+        /* Reformat $toPhone number */
+        $cleanup_chr = array ("+", " ", "(", ")", "\r", "\n", "\r\n");
+        $toPhone = str_replace($cleanup_chr, "", $toPhone);
+        if (!str_starts_with($toPhone, "1")) {
```

Comment:
Because we don't know if this method is being called from API or a send dialog need to test for and resolve:

      $authErrorMsg = $this->authenticate(); // for API calls at minimum
        if ($authErrorMsg !== 1) {
            return text(js_escape($authErrorMsg));
            // goes to alert
        }
       
       $toPhone = $toPhone ?: $this->getRequest('phone');
        $from = $from ?: $this->getRequest('from');
        $message = $message ?: $this->getRequest('comments');

        $toPhone = $this->formatPhoneForSave($toPhone);